src/pipdeptree/_models/package.py

```diff
@@ -51,6 +50,10 @@ def licenses(self) -> str:
 
         return f'({", ".join(license_strs)})'
 
+    @property
+    def project_name(self) -> str:
+        return pep503_normalize(self._project_name)
+
```

Comment:
> The left testcases are all checking `project_name` and `key`. For example
> 
> ```
>         graph: dict[tuple[str, str], list[tuple[str, list[tuple[str, str]]]]] = {
>             ("examplePy", "1.2.3"): [("hellopy", [(">=", "2.0.0")])],
>             ("HelloPy", "2.2.0"): [],
>         }
>         package_dag = PackageDAG.from_pkgs(list(mock_pkgs(graph)))
>         parent_key = "examplepy"
>         c = package_dag.get_children(parent_key)
>         assert len(c) == 1
> >       assert c[0].project_name == "HelloPy"
> E       AssertionError: assert 'hellopy' == 'HelloPy'
> E
> E         - HelloPy
> E         + hellopy
> ```

The reason this is failing is because you are normalizing the project name, which we shouldn't do here.

Reply:
But `pep503_normalize` won't lowercase the project_name, I think it may be caused by the changes elsewhere.

Reply:
It should; looks like it's missing `pep503_normalize` is missing a `.lower()`

Reply:
Sorry, this is actually an old unit test that exists because of some pip weirdness that occurred with requirement names. Since it doesn't look like we have to worry about that any longer:
```console
>>> d.metadata["Name"]
'diff_cover'
>>> d.requires
['Jinja2 (>=2.7.1)', 'Pygments (>=2.9.0,<3.0.0)', 'chardet (>=3.0.0)', 'pluggy (>=0.13.1,<2)', 'setuptools (>=17.0.0) ; python_version < "3.8"', 'tomli (>=1.2.1) ; extra == "toml"']
```
(notice `Jinja2` uses its raw name)

... I think we can remove this unit test.

Reply:
done this